boards/shields/npm1300_ek/npm1300_ek.overlay

```diff
@@ -46,5 +46,27 @@
 				enable-gpios = <&npm1300_ek_gpio 2 GPIO_ACTIVE_LOW>;
 			};
 		};
+
+		npm1300_ek_charger: charger {
+			compatible = "nordic,npm1300-charger";
+			term-microvolt = <4150000>;
+			term-warm-microvolt = <4000000>;
+			current-microamp = <150000>;
+			dischg-limit-microamp = <1000000>;
+			vbus-limit-microamp = <500000>;
+			thermistor-ohms = <10000>;
+			charging-enable;
+			thermistor-lookup = <
+			32767   19672   15455   13288   11857   10797   9959    9267
+			8678    8165    7710    7301    6929    6587    6271    5976
+			5699    5438    5190    4954    4728    4512    4303    4102
+			3906    3717    3532    3352    3175    3002    2832    2665
+			2500    2336    2174    2013    1853    1693    1534    1374
+			1214    1053    891     727     561     392     220     45
+			(-134)  (-318)  (-509)  (-708)  (-914)  (-1131) (-1360) (-1605)
+			(-1867) (-2154) (-2471) (-2830) (-3248) (-3759) (-4431) (-5479)
+			(-32768)
+			>;
```

Comment:
note: considering DT entries for the PMIC will need to be added to every new board, this looks a bit cumbersome from the usability perspective. I feel like we should be dealing with this internally in the driver, ie, use DSP log calls (which I assume will be efficient) or some sort of approximation.